src/memory-analyzer/analyze_symbol.h

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 Author: Malte Mues
+#ifdef __linux__
+#ifndef CPROVER_MEMORY_ANALYZER_ANALYZE_SYMBOL_H
+#define CPROVER_MEMORY_ANALYZER_ANALYZE_SYMBOL_H
+#include <map>
+#include <string>
+
+#include <ansi-c/expr2c_class.h>
+#include <util/message.h>
+#include <util/namespace.h>
+#include <util/std_code.h>
+
+class symbol_tablet;
+class gdb_apit;
+class exprt;
+class source_locationt;
+
+class symbol_analyzert : public messaget
+{
+public:
+  symbol_analyzert(
+    const symbol_tablet &st,
+    gdb_apit &gdb,
+    message_handlert &handler);
+  void analyse_symbol(const std::string &symbol);
+  std::string get_code();
+
+private:
+  gdb_apit &gdb_api;
```

Comment:
Would it be alright to change this so the symbol analyzer owns the GDB API, or will we generally need to do something with it after the analyzer finishes?

Reply:
We do a few things before but nothing that couldn't be encapsulated.